src/main/java/com/aws/greengrass/logmanager/util/SdkClientWrapper.java

```diff
@@ -74,7 +74,7 @@ private void refreshClient() {
     private boolean shouldRefreshClient(SdkClientException e) {
         Throwable cause = e;
         while (cause != null) {
-            if (cause instanceof SocketException && "Connection reset".equals(cause.getMessage())) {
+            if (cause instanceof SocketException && "connection reset".contains(cause.getMessage().toLowerCase())) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code checks the status message of the exception instead of the status itself. You could help improve backwards compatibility by checking the error code, which is less likely to change over time.